src/json.h

```diff
@@ -698,6 +698,46 @@ class JsonObject
             return jsin->read(t);
         }
 
+        template <typename T>
```

Comment:
This doesn't belong in the json module, we might want to wrap it in another class for our own extensions, but for the moment I'm going to make a PR moving these to item_factory.cpp since that's where they're used.